libcloud/compute/drivers/gig_g8.py

```diff
@@ -0,0 +1,501 @@
+        self.driver = driver
+        self.publicipaddress = publicipaddress
+        self.extra = extra
+        UuidMixin.__init__(self)
+
+    @property
+    def cidr(self):
+        """
+        Cidr is not part of the list result
+        we will lazily fetch it with a get request
+        """
+        if self._cidr is None:
+            networkdata = self.driver._api_request("/cloudspaces/get",
+                                                   {"cloudspaceId": self.id})
+            self._cidr = networkdata["privatenetwork"]
+        return self._cidr
+
+    def list_nodes(self):
+        return self.driver.list_nodes(self)
+
+    def destroy(self):
+        return self.driver.ex_destroy_network(self)
+
+    def list_portforwards(self):
+        return self.driver.ex_list_portforwards(self)
+
+    def create_portforward(self, node, publicport,
+                           privateport, protocol='tcp'):
+        return self.driver.ex_create_portforward(self, node, publicport,
+                                                 privateport, protocol)
+
+
+class G8NodeDriver(NodeDriver):
+    """
+    GiG G8 node driver
+
+    """
+
+    NODE_STATE_MAP = {'VIRTUAL': NodeState.PENDING,
+                      'HALTED': NodeState.STOPPED,
+                      'RUNNING': NodeState.RUNNING,
+                      'DESTROYED': NodeState.TERMINATED,
+                      'DELETED': NodeState.TERMINATED,
+                      'PAUSED': NodeState.PAUSED,
+                      'ERROR': NodeState.ERROR,
+
+                      # transition states
+                      'DEPLOYING': NodeState.PENDING,
+                      'STOPPING': NodeState.STOPPING,
+                      'MOVING': NodeState.MIGRATING,
+                      'RESTORING': NodeState.PENDING,
+                      'STARTING': NodeState.STARTING,
+                      'PAUSING': NodeState.PENDING,
+                      'RESUMING': NodeState.PENDING,
+                      'RESETTING': NodeState.REBOOTING,
+                      'DELETING': NodeState.TERMINATED,
+                      'DESTROYING': NodeState.TERMINATED,
+                      'ADDING_DISK': NodeState.RECONFIGURING,
+                      'ATTACHING_DISK': NodeState.RECONFIGURING,
+                      'DETACHING_DISK': NodeState.RECONFIGURING,
+                      'ATTACHING_NIC': NodeState.RECONFIGURING,
+                      'DETTACHING_NIC': NodeState.RECONFIGURING,
+                      'DELETING_DISK': NodeState.RECONFIGURING,
+                      'CHANGING_DISK_LIMITS': NodeState.RECONFIGURING,
+                      'CLONING': NodeState.PENDING,
+                      'RESIZING': NodeState.RECONFIGURING,
+                      'CREATING_TEMPLATE': NodeState.PENDING,
+                      }
+
+    name = "GiG G8 Node Provider"
+    website = 'https://gig.tech'
+    type = Provider.GIG_G8
+    connectionCls = G8Connection
+
+    def __init__(self, apiurl, token, account_id):
```

Comment:
For consistency with the base API, constructor argument should look something like this:

```python
def __init__(self, user_id, key, api_url, ....
```

And in this case, user_id would be account id.